core/src/test/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCOTest.scala

```diff
@@ -466,4 +467,156 @@ class RecordInsightsLOCOTest extends FlatSpec with TestSparkContext {
+
+    val dateMapData: Seq[DateMap] = RandomMap.of(
+      RandomIntegral.dates(refDate.toDate, minStep, maxStep).withProbabilityOfEmpty(0.3),
+      minSize = 0, maxSize = 3
+    ).limit(numRows)
+
+    val dateTimeMapData: Seq[DateTimeMap] = RandomMap.of(
+      RandomIntegral.datetimes(refDate.toDate, minStep, maxStep).withProbabilityOfEmpty(0.3),
+      minSize = 0, maxSize = 3
+    ).limit(numRows)
+
+    val labelData: Seq[RealNN] = RandomIntegral.integrals(0, 2).limit(numRows).map(_.value.get.toRealNN)
+
+    val generatedDateData: Seq[(Date, DateTime, DateMap, DateTimeMap, RealNN)] = dateData.zip(dateTimeData)
+      .zip(dateMapData).zip(dateTimeMapData).zip(labelData)
+      .map { case ((((d, t), dmap), tmap), l) => (d, t, dmap, tmap, l) }
+
+    val (dateDF, date, datetime, dateMap, dateTimeMap, labelNoRes) = TestFeatureBuilder(
+      "date", "datetime", "dateMap", "dateTimeMap", "label", generatedDateData)
+
+    val rawData = dateDF.withColumn("id", monotonically_increasing_id())
+
+    val label = labelNoRes.copy(isResponse = true)
+
+    // Apply date vectorizer
+    val dateVector = date.vectorize(
+      dateListPivot = DateListPivot.SinceLast,
+      trackNulls = true,
+      referenceDate = refDate)
+
+    val datetimeVector = datetime.vectorize(
+      dateListPivot = DateListPivot.SinceLast,
+      trackNulls = true,
+      referenceDate = refDate)
+
+    val dateMapVector = dateMap.vectorize(
+      defaultValue = 0.0,
+      trackNulls = true,
+      referenceDate = refDate
+    )
+
+    val datetimeMapVector = dateTimeMap.vectorize(
+      defaultValue = 0.0,
+      trackNulls = true,
+      referenceDate = refDate
+    )
+
+    val featureVector = Seq(dateVector, datetimeVector, dateMapVector, datetimeMapVector).combine()
+
+    val featureTransformedDF = new OpWorkflow().setResultFeatures(featureVector, label).transform(rawData)
+
+    val sparkModel = new OpLogisticRegression().setInput(label, featureVector).fit(featureTransformedDF)
+
+    // RecordInsightsLOCO
+    val locoTransformer = new RecordInsightsLOCO(sparkModel).setInput(featureVector).setTopK(40)
+
+    val locoInsights = locoTransformer.transform(featureTransformedDF)
+
+    val parsedLocoInsights = locoInsights.collect(locoTransformer.getOutput()).map(i =>
+      RecordInsightsParser.parseInsights(i))
+
+    parsedLocoInsights.foreach(_.values.foreach(a => assert(math.abs(a.map(_._2).sum) < 1e-10, "LOCOs sum to 0")))
+
+    val meta = OpVectorMetadata.apply(featureTransformedDF.schema(featureVector.name))
+
+    implicit val enc: Encoder[(Array[Double], Long)] = ExpressionEncoder()
+    implicit val enc2: Encoder[Seq[Double]] = ExpressionEncoder()
+
+    /**
+     * Compare the aggregation made by RecordInsightsLOCO to one made manually
+     *
+     * @param predicate  predicate used by RecordInsights in order to aggregate
+     */
+    def assertAggregatedWithPredicate(predicate: OpVectorColumnHistory => Boolean): Unit = {
```

Comment:
Same here. Can we reuse the code in the other test to avoid duplication?

Reply:
sure

Reply:
done.